src/lib/pages/administration/Administration.svelte

```diff
@@ -0,0 +1,320 @@
+<script lang="ts">
+  import {
+    Button,
+    ButtonGroup,
+    Dropdown,
+    DropdownItem,
+    Input,
+    Label,
+    Radio,
+    Select,
+    Spinner,
+  } from 'flowbite-svelte';
+
+  import * as Realm from 'realm-web';
+  import { mongo, realmApp } from '../../../main';
+  import { Constants, MongoCollections } from '../../../Constants';
+  import { getAllContexts, onMount } from 'svelte';
+  import type { UserData } from '../../models/UserData/UserData';
+  import {
+    Table,
+    TableBody,
+    TableBodyCell,
+    TableBodyRow,
+    TableHead,
+    TableHeadCell,
+  } from 'flowbite-svelte';
+  import { AccountRole } from '../../enums/AccountRole';
+  import { AccountState } from '../../enums/AccountState';
+  import * as fsPromise from 'fs/promises';
+
+  let isLoading = true;
+  const {
+    BSON: { ObjectId },
+  } = Realm;
+
+  onMount(async () => {
+    await getUsers();
+  });
+
+  var users = new Array<UserData>();
+  let isFormOpen = false;
+
+  async function getUsers() {
+    const data = mongo.collection(MongoCollections.Users);
+    const result = (await data.find()) as UserData[];
+    users = result;
+    isLoading = false;
+  }
+
+  async function download(fileName, contentType) {
+    const users = mongo.collection(MongoCollections.Users);
+    var result = await users.find();
+    const data = JSON.stringify(result);
+    console.log(data);
+
+    var a = document.createElement('a');
+    var file = new Blob([data], { type: contentType });
+    a.href = URL.createObjectURL(file);
+    a.download = fileName;
+    a.click();
+    URL.revokeObjectURL(a.href);
+  }
+
+  async function onSaveUser(user: UserData) {
+    const mongo = realmApp.currentUser.mongoClient(
+      import.meta.env.VITE_DATA_SOURCE_NAME
+    );
+
+    const data = await mongo
+      .db(Constants.DatabaseName)
+      .collection(MongoCollections.Users)
+      .updateOne(
+        { personalId: user.personalId },
+        {
+          $set: {
+            name: user.name,
+            surname: user.surname,
+            nationalIdNumber: user.nationalIdNumber,
+            nickname: user.nickname,
+            email: user.email,
+            address: user.address,
+            role: user.role,
+          },
+        }
+      );
+    getUsers();
+  }
+
+  async function onDeleteUser(user: UserData) {
+    const mongo = realmApp.currentUser.mongoClient(
+      import.meta.env.VITE_DATA_SOURCE_NAME
+    );
+    realmApp.currentUser.customData;
+
+    const data = await mongo
+      .db(Constants.DatabaseName)
+      .collection(MongoCollections.Users)
+      .deleteOne({ personalId: user.personalId });
+    getUsers();
+  }
+
+  async function onChangeActivityStateOfUser(user: UserData) {
+    var state: AccountState;
+    if (user.accountState === 0) {
+      state = 1;
+    } else state = 0;
+
+    const data = await mongo.collection(MongoCollections.Users).updateOne(
+      { personalId: user.personalId },
+      {
+        $set: {
```

Comment:
Můžu změnit state na stávající state? Chybí mi tu check at se nevola be zbytečně třeba